agent/agent.go

```diff
@@ -1061,13 +1068,13 @@ func setEndTaskFailureDetails(tc *taskContext, detail *apimodels.TaskEndDetail,
 
 }
 
-func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupCheck bool, reason string) {
+func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupCheck bool, reason string, additionalShouldKillCheck bool) {
 	logger := grip.NewJournaler("killProcs")
 	if tc.logger != nil && !tc.logger.Closed() {
 		logger = tc.logger.Execution()
 	}
 
-	if !a.shouldKill(tc, ignoreTaskGroupCheck) {
+	if !a.shouldKill(tc, ignoreTaskGroupCheck) && !additionalShouldKillCheck {
```

Comment:
1. If the original problem was that `a.shouldKill` dereferences a nil `taskConfig` but this PR already changed the logic to check for nil `taskConfig`, is the `additionalShouldKillCheck` flag still needed? It seems specially skipping `shouldKill` wouldn't matter the nil `taskConfig` check is already in place.
2. Is the goal of this flag to skip running `a.shouldKill` specifically when `makeTaskConfig` fails? If so, the two clauses in this if statement have to be swapped (to `!additionalShouldKillCheck && !a.shouldKill(tc, ignoreTaskGroupCheck)`) because Go evaluates each `&&`-separated clause in order.

Reply:
It doesn't need to be swapped because there's no longer a nil pointer dereference issue. a.shouldkill is not a heavy process so there's no need to save that unit of work in the case where !additionalShouldKillCheck  is false.